flight/PiOS/posix/pios_thread.c

```diff
@@ -155,7 +155,21 @@ static volatile uint32_t fake_tick_barrier;
 
 void PIOS_Thread_FakeClock_UpdateBarrier(uint32_t increment)
 {
+	pthread_mutex_lock(&fake_clock_mutex);
```

Comment:
Use the condvar here to improve the wakeup behavior instead of sleeping a short quanta.